Use actual mod-adjusted map difficulty settings in the `SongBar`

stanriders opened the pull request:
This also changes the bpm format to be consistent with the rest of the game. I would love for the SR display to show real values too, but that requires loading a full beatmap and the tournament client only has access to the metadata.

NM:
<img width="458" height="122" alt="image" src="https://github.com/user-attachments/assets/03720049-2b9b-45c1-a92e-21319de8a639" />

HR:
<img width="462" height="113" alt="image" src="https://github.com/user-attachments/assets/a8f7cff1-a457-4929-8299-ca4d710dd2ed" />

DT:
<img width="457" height="110" alt="image" src="https://github.com/user-attachments/assets/fcb61098-0c4b-4d4d-97f0-62fac90d72dc" />

Mania NM:
<img width="456" height="112" alt="image" src="https://github.com/user-attachments/assets/6d4a9319-6ea1-417f-b3a1-5d87ceb903eb" />

Mania HR:
<img width="458" height="108" alt="image" src="https://github.com/user-attachments/assets/427cd7b6-c1f3-4ec8-b2ab-6ab53a543555" />

stanriders committed:
Use actual mod-adjusted map difficulty settings in the `SongBar`

stanriders committed:
Add ruleset tests

stanriders committed:
Set `TournamentBeatmap`'s `IBeatmapInfo.Ruleset` to a dummy ruleset.

This is being queried by the https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Mania/ManiaRuleset.cs#L442 but since we don't actually draw column count anywhere nor are we supposed to be running converts in tournaments it should be safe to populate it with nothing.

peppy commented:
Could probably download the `.osu` file to get full stats using the `/osu/` endpoint?

stanriders commented on osu.Game.Tournament/Models/TournamentBeatmap.cs:
```diff
@@ -83,7 +83,7 @@ public TournamentBeatmap(APIBeatmap beatmap)
         string IBeatmapInfo.MD5Hash => throw new NotImplementedException();
 
-        IRulesetInfo IBeatmapInfo.Ruleset => throw new NotImplementedException();
+        IRulesetInfo IBeatmapInfo.Ruleset => new RulesetInfo();
```
<img width="1294" height="164" alt="image" src="https://github.com/user-attachments/assets/82bf7362-ad7f-4960-a314-9528cb8610d6" />
TournamentBeatmap by itself is pretty much just a metadata wrapper

bdach replied:
I don't get why though, surely you could just

```diff
diff --git a/osu.Game.Tournament/Models/TournamentBeatmap.cs b/osu.Game.Tournament/Models/TournamentBeatmap.cs
index a7ba5b7db1..59003bc739 100644
--- a/osu.Game.Tournament/Models/TournamentBeatmap.cs
+++ b/osu.Game.Tournament/Models/TournamentBeatmap.cs
@@ -47,6 +47,7 @@ public TournamentBeatmap(APIBeatmap beatmap)
             Covers = beatmap.BeatmapSet?.Covers ?? new BeatmapSetOnlineCovers();
             EndTimeObjectCount = beatmap.EndTimeObjectCount;
             TotalObjectCount = beatmap.TotalObjectCount;
+            Ruleset = beatmap.Ruleset;
         }
 
         public bool Equals(IBeatmapInfo? other) => other is TournamentBeatmap b && this.MatchesOnlineID(b);

```

no? What am I missing here?

stanriders committed:
Use `APIRuleset` instead of a blank `RulesetInfo`

bdach committed:
Remove unnecessary cast

bdach committed:
Fix test failures

This is dodgy as hell but `ShortName` is completely derived from
`OnlineID` anyway so there should be no valid reason to ever attempt to
serialise it anyway.
